frontend/src/components/common/ActionsNotifier.tsx

```diff
@@ -49,13 +50,28 @@ function PureActionsNotifier({ dispatch, clusterActions }: PureActionsNotifierPr
       closeSnackbar(clusterAction.dismissSnackbar);
     }
 
+    const prevKey = snackbarRefs.current[clusterAction.id];
+    const uniqueKey = clusterAction.key || clusterAction.id;
+
+    if (prevKey && prevKey !== uniqueKey) {
+      closeSnackbar(prevKey);
+    }
+
     if (clusterAction.message) {
-      enqueueSnackbar(clusterAction.message, {
-        key: clusterAction.key,
-        autoHideDuration: clusterAction.autoHideDuration || CLUSTER_ACTION_GRACE_PERIOD,
-        action,
-        ...clusterAction.snackbarProps,
-      });
+      // Check for completed actions
+      const refKey = !clusterAction.message.includes('ing')
```

Comment:
the `.includes('ing')` seems way too hacky, is there other way to achieve this?

Reply:
It covers all the current cases, but you are right that it is not fool proof. The alternative would be to add a new state to ClusterAction to keep track of what stage it is in since this is currently not implemented, can make this change though